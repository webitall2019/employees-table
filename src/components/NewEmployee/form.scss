.form__wrap{
    padding: 120px;
    width: 80%;
    background-color: #fff;
    h1{
        text-align: center;
    }
}
.form__group{
    display: flex;
    max-width: 60%;
    margin: 0 auto;
    flex-direction: column;
    align-items: stretch;
    justify-content: space-between;
    height: 100%;
    input, select{
       
        padding: 10px;
        margin-bottom: 20px;
    }
    @media screen and (max-width: 500px) {
        max-width: initial;
    }
}