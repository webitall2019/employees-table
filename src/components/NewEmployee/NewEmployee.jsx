import { useDispatch } from "react-redux";
import React, { useState, useEffect } from "react";
import "./form.scss";

const NewEmployee = () => {
  const [newEmployee, setNewEmployee] = useState({});
  const [name, setName] = useState("");
  const [tenure, setAge] = useState("");
  const [jobTitle, setJob] = useState("CEO");
  const [gender, setGender] = useState("Male");
  const dispatch = useDispatch();

  useEffect(() => {
    setNewEmployee({
      name,
      tenure,
      jobTitle,
      gender,
    });
  }, [name, tenure, jobTitle, gender]);

  const submithandler = async (e) => {
    e.preventDefault();
    dispatch({ type: "newUser", payload: newEmployee });
  };

  return (
    <div className="form__wrap">
      <h1>Become one of the team</h1>
      <form onSubmit={submithandler}>
        <div className="form__group">
          <input
            data-testid="userName"
            required
            placeholder="your name"
            id="name"
            type="text"
            name="name"
            value={name}
            onChange={(e) => setName(e.target.value)}
          />
          <input
            data-testid="userAge"
            required
            placeholder="your age"
            type="number"
            name="tenure"
            value={tenure}
            onChange={(e) => setAge(e.target.value)}
          />
          <select
            data-testid="jobTitle"
            name="jobTitle"
            onChange={(e) => setJob(e.target.value)}
            required
          >
            <option value="CEO">CEO</option>
            <option value="Developer">Developer</option>
            <option value="Marketing Specialist">Marketing Specialist</option>
            <option value="CFO">CFO</option>
            <option value="QA">QA</option>
          </select>
          <select
            data-testid="gender"
            name="gender"
            onChange={(e) => setGender(e.target.value)}
            required
          >
            <option value="Male">Male</option>
            <option value="Female">Female</option>
          </select>
          <button type="submit">submit</button>
        </div>
      </form>
    </div>
  );
};
export default NewEmployee;
