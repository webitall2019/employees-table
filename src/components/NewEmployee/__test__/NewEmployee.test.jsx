import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import NewEmployee from "../NewEmployee";
import store from "../../../store/store";
import { Provider } from "react-redux";

describe("<NewEmployee/> component", () => {
  it("input user name", () => {
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <NewEmployee />
      </Provider>
    );

    const inputUserName = screen.getByTestId("userName");
    const inputUserAge = screen.getByTestId("userAge");
    const selectUserjobTitle = screen.getByTestId("jobTitle");
    const selectUserGender = screen.getByTestId("gender");

    fireEvent.change(inputUserName, { target: { value: "Bob" } });
    fireEvent.change(inputUserAge, { target: { value: "25" } });
    fireEvent.change(selectUserjobTitle, { target: { value: "CEO" } });
    fireEvent.change(selectUserGender, { target: { value: "Female" } });

    expect(inputUserName.value).toBe("Bob");
    expect(inputUserAge.value).toBe("25");

    expect(selectUserjobTitle.value).toBe("CEO");
    expect(selectUserGender[0].selected).toBeFalsy();
    expect(selectUserGender[1].selected).toBeTruthy();
  });

  it("should match snapshot WHEN rendered", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <NewEmployee />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
