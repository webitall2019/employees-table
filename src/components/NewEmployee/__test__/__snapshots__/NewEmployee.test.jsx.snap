// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewEmployee/> component should match snapshot WHEN rendered 1`] = `
<DocumentFragment>
  <div
    class="form__wrap"
  >
    <h1>
      Become one of the team
    </h1>
    <form>
      <div
        class="form__group"
      >
        <input
          data-testid="userName"
          id="name"
          name="name"
          placeholder="your name"
          required=""
          type="text"
          value=""
        />
        <input
          data-testid="userAge"
          name="tenure"
          placeholder="your age"
          required=""
          type="number"
          value=""
        />
        <select
          data-testid="jobTitle"
          name="jobTitle"
          required=""
        >
          <option
            value="CEO"
          >
            CEO
          </option>
          <option
            value="Developer"
          >
            Developer
          </option>
          <option
            value="Marketing Specialist"
          >
            Marketing Specialist
          </option>
          <option
            value="CFO"
          >
            CFO
          </option>
          <option
            value="QA"
          >
            QA
          </option>
        </select>
        <select
          data-testid="gender"
          name="gender"
          required=""
        >
          <option
            value="Male"
          >
            Male
          </option>
          <option
            value="Female"
          >
            Female
          </option>
        </select>
        <button
          type="submit"
        >
          submit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
